src/Components/Medicine/MedicineAdministrationSheet/index.tsx

```diff
@@ -97,47 +101,66 @@ const MedicineAdministrationSheet = ({ readonly, is_prn }: Props) => {
           )
         }
       />
-
-      <ScrollOverlay
-        className="rounded-lg border shadow"
-        overlay={
-          <div className="flex items-center gap-2 pb-2">
-            <span className="text-sm">Scroll to view more prescriptions</span>
-            <CareIcon icon="l-arrow-down" className="animate-bounce text-2xl" />
-          </div>
-        }
-        disableOverlay={loading || !prescriptions?.length}
-      >
-        {loading && <Loading />}
-        {prescriptions?.length === 0 && <NoPrescriptions prn={is_prn} />}
-
-        {!!prescriptions?.length && (
-          <MedicineAdministrationTable
-            prescriptions={prescriptions}
-            pagination={pagination}
-            onRefetch={() => {
-              refetch();
-              discontinuedPrescriptions.refetch();
-            }}
-          />
-        )}
-
-        {!showDiscontinued && !!discontinuedCount && (
+      <div className="rounded-lg border shadow">
+        <ScrollOverlay
+          overlay={
+            <div className="flex items-center gap-2 pb-2">
+              <span className="text-sm">Scroll to view more prescriptions</span>
+              <CareIcon
+                icon="l-arrow-down"
+                className="animate-bounce text-2xl"
+              />
+            </div>
+          }
+          disableOverlay={
+            loading || !prescriptions?.length || !(prescriptions?.length > 1)
+          }
+        >
+          {loading ? (
+            <Loading />
+          ) : (
+            <>
+              {prescriptions?.length === 0 && <NoPrescriptions prn={is_prn} />}
+              {!!prescriptions?.length && (
+                <MedicineAdministrationTable
+                  prescriptions={prescriptions}
+                  pagination={pagination}
+                  onRefetch={() => {
+                    refetch();
+                    discontinuedPrescriptions.refetch();
+                  }}
+                />
+              )}
+            </>
+          )}
+        </ScrollOverlay>
+        {!!discontinuedCount && (
           <ButtonV2
             variant="secondary"
             className="group sticky left-0 w-full rounded-b-lg rounded-t-none bg-gray-100"
-            onClick={() => setShowDiscontinued(true)}
+            onClick={() => !loading && setShowDiscontinued(!showDiscontinued)}
```

Comment:
```suggestion
            disable={discontinuedPrescriptions.loading}
            onClick={() => setShowDiscontinued(!showDiscontinued)}
```

Reply:
here if i keep `discontinuedPrescriptions.loading` instead of `loading` then the button is not disabled in between of show and hide timing

Reply:
![image](https://github.com/coronasafe/care_fe/assets/98876115/818ef906-080a-45f1-aa01-362256eef389)

Reply:
Then you can disable based on both loading states. 

This button is dependent on the discontinued prescriptions query and not the active prescriptions query so depending on discontinued prescriptions loading state is necessary.